src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,82 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        ArrayList<TimeRange> returnList = new ArrayList<TimeRange>();
+        returnList.add(TimeRange.WHOLE_DAY);
+
+        trimList(returnList, request.getAttendees(), events);
+
+        removeSmallSlots(returnList, request.getDuration());
+
+        return returnList;
+    }
+
+    private void removeSmallSlots(ArrayList<TimeRange> timeList, long duration) {
```

Comment:
"Parameters are most naturally interpreted as inputs to a function. As a result, output parameters are unexpected, which makes code harder to understand at first glance."

timeList is used as an output parameter here. A better approach would be to not mutate the parameter but to create a new list and return it, similar to Java's String.trim() method. If you do that, you wouldn't have to call clone, which would actually make your method more performant.

So you would call this method as follows:
`returnList = removeSmallSlots(returnList, request.getDuration());`